src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -625,6 +625,11 @@ public Claim GetClaim(string key)
             return Payload.GetClaim(key, Issuer ?? ClaimsIdentity.DefaultIssuer);
         }
 
+        /// <summary>
+        /// Gets the keys on the JsonWebToken.
+        /// </summary>
+        public IReadOnlyCollection<string> Keys => Payload._jsonClaims.Keys;
```

Comment:
Do we need to make this public as it is only used internally.
What can a user do with this value?

Reply:
Name might be a bit misleading as we normally think of keys as a securitykey.
Also, these keys are only from the payload.

Reply:
JsonWebToken isn't internals visible to Validators. I'd really like to avoid it.

The utility is the same as we are using it for, if you want to look at the types of the claims without creating the claims.

Happy to rename if we can come up with a better name

Reply:
I agree on the naming being a bit misleading. Perhaps PayloadClaimNames?